docs2/site/docs/migrations/migration4.md

```diff
@@ -278,6 +278,165 @@ lock (field)
+}
+
+public class MoneyType : ObjectGraphType<Money>
+{
+    public MoneyType()
+    {
+        Field(x => x.Amount);
+        Field(x => x.Currency);
+    }
+}
+
+public class MoneyInputType : InputObjectGraphType<Money>
+{
+    public MoneyInputType()
+    {
+        Field(x => x.Amount).Description("Total amount").DefaultValue(100m);
+        Field(x => x.Currency).DefaultValue("USD");
+    }
+}
+
+schema.RegisterTypeMapping<Money, MoneyType>();
+schema.RegisterTypeMapping<Money, MoneyInputType>();
+```
+
+An alternative way to define the mapping is to use the new properties in the `GraphQLMetadata` attribute.
+Consider the following example:
+
+```c#
+[GraphQLMetadata(InputType = typeof(FilterInputGraphType))]
+public class Filter
+{
+    public string Key { get; set; }
+    public int Value { get; set; }
+}
+
+public class ContainerRequest
+{
+    public IList<Filter> Filters { get; set; }
+    public int ClientId { get; set; }
+    public int AppId { get; set; }
+}
+
+public class FilterInputGraphType : InputObjectGraphType<Filter>
+{
+    public FilterInputGraphType()
+    {
+        Name = "FilterInput";
+        Field(x => x.Key);
+        Field(x => x.Value);
+    }
+}
+
+public class MyInputType : InputObjectGraphType<ContainerRequest>
+{
+    public MyInputType()
+    {
+        Name = "Input";
+        Field(x => x.Filters); // when building this field, its type is implicitly inferred to list of FilterInputGraphType
+        Field(x => x.ClientId);
+        Field(x => x.AppId, nullable: true);
+    }
+}
+```
+
+In this case, a call to the registration method is not required, since the schema
+will use information from the provided attribute.
+
+> Keep in mind that you can register type mappings even for built-in/primitive types if you want to change their behavior:
+> <br/><br/>
+> schema.RegisterTypeMapping<int, MyIntGraphType>()
+> <br/>
+> schema.RegisterTypeMapping<string, MySpecialFormattedStringGraphType>()
+
+### Classes for automatic GraphType registration by default use all properties of the CLR type
+
+In v4 `AutoRegisteringObjectGraphType<TSourceType>` and `AutoRegisteringInputObjectGraphType<TSourceType>`
+classes by default use all properties from the provided `TSourceType` to generate GraphType's fields. If no matching
+is found for some of the properties, then an exception will be thrown during schema initialization. You need to register
+the type mapping with `ISchema.RegisterTypeMapping` method. Alternatively, you can inherit from these classes and
+override the `GetRegisteredProperties` method. Consider the following example:
+
+```csharp
+public class MyAutoType : AutoRegisteringObjectGraphType<SomeClassWithManyProperties>
+{
+    protected override IEnumerable<PropertyInfo> GetRegisteredProperties() => typeof(SomeClassWithManyProperties)
+        .GetProperties(BindingFlags.Public | BindingFlags.Instance).Where(p => Attribute.IsDefined(p, typeof(ForExportAttribute)));
+}
+```
```

Comment:
This explains current/new behavior very well, and how the `GetRegisteredProperties` method works, but doesn't explain why it's a breaking change, or how to fix old code so it will compile/run under v4.  It should say that previously it would skip unmatched properties, and then provide a sample of how to fix old code.  In this case, the sample should show how to pass the unmatchable property into the `excludedProperties` field of the constructor.  That, I would think, is the simplest fix for someone with properties that should not be matched.

Reply:
done

Reply:
After the v4 release, we will need to copy all of this (with some edits) into the main documentation.